File: experiment/switch_ifelse_dict/switch.js
function apply_tuple(action, num) {
  var result;

  switch(action){
    case 'half':
      result = num / 2;
      break;
    case 'double':
      result = num * 2;
      break;
    case 'triple':
      result = num * 3;
      break;
    case 'quadruple':
      result = num * 4;
      break;
    default: 
      result = num;
  }

  return result;
}

const result = apply_tuple('double', 2);

File: experiment/switch_ifelse_dict/switch.rb
def apply_tuple(action, num)

  case action
    when :half
      result = num / 2;
    when :double
      result = num * 2;
    when :triple
      result = num * 3;
    when :quadruple
      result = num * 4;
    else
      result = num;
  end

  return result
end

result = apply_tuple :double, 2
